docs/building-a-node-with-ldk/setting-up-a-channel-manager.md

```diff
@@ -0,0 +1,1276 @@
+channelManager.update_best_block(bestHeader, bestHeight);
+chainMonitor.update_best_block(bestHeader, bestHeight);
+
+// Finally, tell LDK that chain sync is complete. This will also spawn several
+// background threads to handle networking and event processing.
+channelManagerConstructor.chain_sync_completed(customEventHandler);
+````
+
+  </template>
+
+  <template v-slot:swift>
+
+```Swift
+// Electrum/Esplora
+
+// Retrieve transaction IDs to check the chain for un-confirmation.
+let relevantTxIds1 = channelManager?.asConfirm().getRelevantTxids() ?? []
+let relevantTxIds2 = chainMonitor?.asConfirm().getRelevantTxids() ?? []
+
+var relevantTxIds: [[UInt8]] = [[UInt8]]()
+for tx in relevantTxIds1 {
+    relevantTxIds.append(tx.0)
+}
+for tx in relevantTxIds2 {
+    relevantTxIds.append(tx.0)
+}
+
+var unconfirmedTx: [[UInt8]] = // Insert code to find out from your chain source
+                              // if any of relevantTxIds have been reorged out
+                              // of the chain
+
+for txid in unconfirmedTx {
+    channelManager.asConfirm().transactionUnconfirmed(txid: txid)
+    chainMonitor.asConfirm().transactionUnconfirmed(txid: txid)
+}
+
+// Retrieve transactions and outputs that were registered through the `Filter` interface.
+
+var confirmedTx: [[UInt8]] = // Insert code to find out from your chain source
+                            // if any of the `Filter` txs/outputs were confirmed
+                            // on-chain
+
+for txid in confirmedTx {
+    let header: [UInt8] = // Insert code to fetch header
+    let height: UInt32 = // Insert code to fetch height of the header
+    let tx: [UInt8] = // Insert code to fetch tx
+    let txIndex: UInt = // Insert code to fetch tx index
+
+    var twoTuple: [(UInt, [UInt8])] = []
+    twoTuple.append((UInt, [UInt8])(txIndex, tx))
+    channelManager.asConfirm().transactionsConfirmed(header: header, txdata: twoTuple, height: height)
+    chainMonitor.asConfirm().transactionsConfirmed(header: header, txdata: twoTuple, height: height)
+}
+
+let bestHeader = // Insert code to fetch best header
+let bestHeight = // Insert code to fetch best height
+
+channelManager.asConfirm().bestBlockUpdated(header: bestHeader, height: bestHeight)
+chainMonitor.asConfirm().bestBlockUpdated(header: bestHeader, height: bestHeight)
+
+// Finally, tell LDK that chain sync is complete. This will also spawn several
+// background threads to handle networking and event processing.
+channelManagerConstructor.chainSyncCompleted(persister: channelManagerPersister)
+```
+
+  </template>
+
+</CodeSwitcher>
+
+**Implementation notes:**
+
+There are 2 main options for synchronizing to chain on startup:
+
+**Full Blocks or BIP 157/158**
+
+If you are connecting full blocks or using BIP 157/158, then it is recommended to use
+LDK's `lightning_block_sync` sample module as in the example above: the high-level steps that must be done for both `ChannelManager` and each `ChannelMonitor` are as follows:
```

Comment:
s/`lightning_block_sync` sample module/`lightning-block-sync` crate

Reply:
Fixed in https://github.com/lightningdevkit/lightningdevkit.org/pull/156/commits/74a2f492a2d73de20d0f62f1321d7342523a9382